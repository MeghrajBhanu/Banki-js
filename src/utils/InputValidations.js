export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}