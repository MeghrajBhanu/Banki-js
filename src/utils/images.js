import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;