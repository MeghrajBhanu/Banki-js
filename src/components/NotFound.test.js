import { render, screen, waitFor } from "@testing-library/react";
import { onlywrap, renderComponent } from "../utils/test_helper";
import renderer from "react-test-renderer";
import NotFound from "./NotFound";

describe("NotFound", () => {
  test("snapshot test", () => {
    const tree = renderer.create(onlywrap(<NotFound />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});