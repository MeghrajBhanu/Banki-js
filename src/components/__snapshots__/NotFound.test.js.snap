// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound snapshot test 1`] = `
<div
  className="d-flex align-items-center justify-content-center vh-100"
>
  <div
    className="text-center"
  >
    <h1
      className="display-1 fw-bold mx-auto"
    >
      404
    </h1>
    <p
      className="fs-3"
    >
       
      <span
        className="text-danger"
      >
        Opps!
      </span>
       Page not found.
    </p>
    <p
      className="lead"
    >
      The page you’re looking for doesn’t exist.
    </p>
    <a
      className="btn btn-primary"
      href="/"
      onClick={[Function]}
    >
      Go Home
    </a>
  </div>
</div>
`;
